src/gt4py/next/common.py

```diff
@@ -592,6 +593,16 @@ def __getstate__(self) -> dict[str, Any]:
 )  # `Domain` is `Sequence[NamedRange]` and therefore a subset
 
 
+def normalize_domains(domain_like: MaybeNestedInTuple[DomainLike]) -> MaybeNestedInTuple[Domain]:
```

Comment:
absorb in `domain` (and remove `if isinstance(domain_like, Sequence)` case)?

Reply:
After implementing this [here](https://github.com/GridTools/gt4py/pull/2319) I think this is not a good idea anymore. We need to distinguish between places where only a single domain is allowed, e.g. `allocate`, and places where a tuple of domain likes, e.g. all places where this `normalize_domains` is used. It feels simpler to just do `tree_map(domain)(nested_domain_like)` in all places where `normalize_domains` is used and then to remove this function. Then it is clear that in all other places a single domain is allowed and here multiple.